### PR TITLE
Enable float16 for GpuGemmBatch.

### PR DESCRIPTION
**Pull request by abergeron**
This requires https://github.com/Theano/libgpuarray/pull/447

### CODE REVIEW

File: theano/gpuarray/blas.py [hunk 1]
```diff
@@ -1726,9 +1735,14 @@ def local_inplace_gpuagemm(node, inputs):
 def local_inplace_gpuager(node, inputs):
     return [gpuger_inplace(*inputs)]
 
+@inplace_allocempty(GpuGemmBatch, 0)
```
Comment (lamblin) [inline theano/gpuarray/blas.py]: E302 expected 2 blank lines, found 1

File: theano/gpuarray/blas.py [hunk 2]
```diff
@@ -1726,9 +1735,14 @@ def local_inplace_gpuagemm(node, inputs):
...
+def local_inplace_gpuagemmbatch(node, inputs):
```
Comment (nouiz) [inline theano/gpuarray/blas.py]: This opt would need a simple test. I think current test won't fail if we disable this opt.

File: theano/gpuarray/tests/test_blas.py [hunk 3]
```diff
@@ -130,7 +130,12 @@ def shared(val):
     ("test_b%im%ik%in%i" % (b, m, k, n),
      [rand(b, m, n), rand(), rand(b, m, k), rand(b, k, n), rand()])
     for b, m, k, n in itertools.combinations([2, 3, 5, 7, 11, 13], 4))
-# float16 not supported
+
+gemm_batched_tests['float16'] = [rand(3, 4, 7).astype('float16'),
```
Comment (nouiz) [inline theano/gpuarray/tests/test_blas.py]: This test fail in DebugMode.

Comment (lamblin) [inline theano/gpuarray/tests/test_blas.py]: Do you have the error message?

**Comment by nouiz** [comment]
I lost it, but there was inf or -inf in the computation.

**Comment by slefrancois** [comment]
jenkins test this

File: theano/gpuarray/blas.py (hunk 1 above)
Comment (nouiz) [inline theano/gpuarray/blas.py]: done

File: theano/gpuarray/blas.py (hunk 2 above)
Comment (nouiz) [inline theano/gpuarray/blas.py]: done

File: theano/gpuarray/tests/test_blas.py (hunk 3 above)
Comment (nouiz) [inline theano/gpuarray/tests/test_blas.py]: fixed

**Comment by nouiz** [comment]
travis section 11 failed with this:
```
test_filter_dilation (theano.tensor.nnet.tests.test_corr3d.TestCorr3D) ... 

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```
I restarted it

**Comment by nouiz** [comment]
jenkins error are strange: http://earlgrey.iro.umontreal.ca:8080/job/Theano_PR/133/testReport/junit/theano.tensor.nnet.tests.test_abstract_conv/TestAbstractConvNoOptim/test_all__8__1__6__6____5__1__2__2____1__1____0__0___True__False_/

jenkins test this

**Comment by nouiz** [comment]
jenkins failed again. I have the impression that it don't test the right branch. The error he have is related to another PR, not this one:

http://earlgrey.iro.umontreal.ca:8080/job/Theano_PR/140/execution/node/8/log/

jenkins test this

**Comment by slefrancois** [comment]
jenkins test this

**Comment by lamblin** [comment]
Jenkins test this please.

**Comment by nouiz** [comment]
Jenkins test this please.

**Comment by abergeron** [comment]
This seems like a problem with the reuse of the Theano cache.

**Comment by slefrancois** [comment]
jenkins test this

**Comment by nouiz** [comment]
jenkins test this

**Pull request merged**